echo_handlers.go

```diff
@@ -140,12 +183,22 @@ func (svc *Service) AppsCreateHandler(c echo.Context) error {
 		pairingPublicKey, _ = nostr.GetPublicKey(pairingSecretKey)
 	} else {
 		pairingPublicKey = c.FormValue("pubkey")
+		//validate public key
+		decoded, err := hex.DecodeString(pairingPublicKey)
+		if err != nil || len(decoded) != 32 {
+			svc.Logger.Errorf("Invalid public key format: %s", pairingPublicKey)
+			return c.Redirect(302, "/apps")
+		}
 	}
 
 	err = svc.db.Model(&user).Association("Apps").Append(&App{Name: name, NostrPubkey: pairingPublicKey})
 	if err == nil {
-		pairingUri := template.URL(fmt.Sprintf("nostrwalletconnect://%s?relay=%s&secret=%s", svc.cfg.IdentityPubkey, svc.cfg.Relay, pairingSecretKey))
+		if c.FormValue("returnTo") != "" {
+			return c.Redirect(302, c.FormValue("returnTo"))
+		}
+		pairingUri := template.URL(fmt.Sprintf("nostr+walletconnect://%s?relay=%s&secret=%s", svc.cfg.IdentityPubkey, svc.cfg.Relay, pairingSecretKey))
```

Comment:
We cannot use the new scheme yet because we would break the old Amethyst clients. So please change this back to nostrwalletconnect://

Reply:
true, how does the path forward look here? at some point we break it.

Reply:
Just wait for a month or so.